### DISCUSSION TITLE
How to enable TCP keepalive feature?

### DISCUSSION THREAD

**Discussion by thune-xiaobai** [encode/httpcore#490] [Q&A]
In `urllib3` docs, there're ways to add tcp keepalive options to `HTTPConnection`
```
HTTPConnection.default_socket_options + [
    (socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1),
]
```

but on `httpx` or `httpcore`, I didn't find the way to do same thing, any solutions if I want to enable tcp keepalive?

**Comment by lovelydinosaur** [comment]
There isn't *currently* any API for socket options, no.

There's ways you'd be able to dig into it. The ["backends" are the point at which the network comms actually happen](https://github.com/encode/httpcore/blob/master/httpcore/backends/sync.py#L74) so those are the place to look to start with. We could probably consider some functionality around this for `httpx` at some point. If we get other requests about this then I think we should consider opening an issue for it.
